### ISSUE TITLE
Allow string input to request for pull component/requests

### ISSUE THREAD

**Issue by rwemay** [BHoM/BHoM_Adapter#227]
#### Description:

Dum dum duuuuum.... shoot me down instantly @alelom , but is it crazy to allow just a string input with the functionality of the FilterRequest(type, tag) with a input string? That way when you get used to knowing what objects you want you can just type in the name direct. 

Close issue if mad!

**Comment by alelom** [comment]
That's something for Reflection_Engine, but sure, we can do it! However, can't really see much improvement over what we have already:
![image](https://user-images.githubusercontent.com/6352844/81091874-b9977a80-8ef7-11ea-9a1b-270a9c48adbd.png)

**Comment by alelom** [comment]
What you're saying is instead you would like to allow this?
![image](https://user-images.githubusercontent.com/6352844/81092111-08451480-8ef8-11ea-878b-ad9c4b6b550c.png)

**Comment by rwemay** [comment]
Yep - the latter. Apologies did not realise you could plug type directly in without tag! #keepuprob! 

It looks like something we _should_ enable in any case given the warning above - better to cast to type than return custom, no?

**Comment by al-fisher** [comment]
I would like this.
A. B. & C. work below - would be so handy if D. also worked. 😄 
![image](https://user-images.githubusercontent.com/15924573/81103890-970e5d00-8f09-11ea-9279-a7644965b4ea.png)
